tools/inventory-extract-windows/main.go

```diff
@@ -0,0 +1,175 @@
+
+func main() {
+	flag.Parse()
+	if *helpFlag {
+		fmt.Println("This program do export of INVENTORY partition from EVE drive")
+		fmt.Println("You can see list of available drives with -list flag")
+		fmt.Println("Then you must set Index of drive to -drive flag")
+		fmt.Println("With -dir flag you can provide directory to save files into. " +
+			"The program will create directory in temp if you will not provide this flag.")
+		os.Exit(0)
+	}
+	if *driveNumber == 0 {
+		fmt.Println("Please provide -drive flag with Index of device from the list below:")
+		*drivesList = true
+	}
+	if *drivesList {
+		res, err := exec.Command("wmic", "diskdrive",
+			"where", "MediaType='Removable Media'",
+			"get", "index,caption").CombinedOutput()
+		if err != nil {
+			log.Println(err)
+		} else {
+			fmt.Println(string(res))
+		}
+		time.Sleep(sleepTime)
+		os.Exit(0)
+	}
+	fd0, err := os.Open(`\\.\PhysicalDrive0`)
+	if err != nil {
+		log.Println("This program must be run by Administrator")
+		log.Println("We will ask for permissions")
+		time.Sleep(sleepTime)
+		runMeElevated()
+		os.Exit(0)
+	}
+	_ = fd0.Close()
+	f, err := os.Open(fmt.Sprintf(`\\.\PhysicalDrive%d`, *driveNumber))
+	if err != nil {
+		log.Println(err)
+		time.Sleep(sleepTime)
+		os.Exit(1)
+	}
+	defer f.Close()
+
+	table, err := partition.Read(f, 512, 512)
+
+	if err != nil {
+		log.Println(err)
+		time.Sleep(sleepTime)
+		os.Exit(1)
+	}
+	partitions := table.GetPartitions()
+	found := false
+	for _, el := range partitions {
+		fsCurrent, err := fat32.Read(f, el.GetSize(), el.GetStart(), 512)
+		if err != nil {
+			continue
+		}
+		if fsCurrent.Label() == "INVENTORY" {
+			found = true
+			var dir string
+			if *outputPath != "" {
+				dir, err = filepath.Abs(*outputPath)
+				if err != nil {
+					log.Println(err)
+					continue
+				}
+				err = os.MkdirAll(dir, os.ModeDir)
+				if err != nil {
+					log.Println(err)
+					continue
+				}
+			} else {
+				dir, err = ioutil.TempDir("", "INVENTORY")
+				if err != nil {
+					log.Println(err)
+					continue
+				}
+			}
+			log.Printf("Will save to directory: %s", dir)
+			err = iterateAndSave(fsCurrent, dir, "/")
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+		}
+	}
+	if found {
+		log.Println("Processing done")
+	} else {
+		log.Println("No INVENTORY found")
+	}
+	time.Sleep(sleepTime)
+}
+
+func iterateAndSave(fsCurrent *fat32.FileSystem, dirToSave, dir string) error {
+	d, err := fsCurrent.ReadDir(dir)
+	if err != nil {
+		return err
+	}
+	for _, curFileInfo := range d {
+		if curFileInfo.Name() == "." || curFileInfo.Name() == ".." {
+			continue
+		}
+		log.Println(curFileInfo.Name())
+		if curFileInfo.IsDir() {
+			curDirPath := filepath.Join(dirToSave, curFileInfo.Name())
+			err = os.Mkdir(curDirPath, os.ModeDir)
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+			err = iterateAndSave(fsCurrent, filepath.Join(dirToSave, curFileInfo.Name()), fmt.Sprintf("%s/%s", dir, curFileInfo.Name()))
```

Comment:
I would limit the depth of this recursion. Fat32 can't have symlinks, but somebody might be screwing with Eve by giving too many nesting dir, and trying to make the system crash, or something

Reply:
> And general question. Why not ask windows to mount and use the drive natively.
> 
> Also I am missing some context. What is this INVENTORY about. Is this program supposed to run inside a Windows guest?

I put readme section inside my PR. It is the partition on installation media, where EVE puts some data during installation. After installation we want to extract data from it (USB stick). Our partition table is not recognized by Windows natively.

Reply:
> I would limit the depth of this recursion. Fat32 can't have symlinks, but somebody might be screwing with Eve by giving too many nesting dir, and trying to make the system crash, or something

You are right here, but it must be used on our installation media with INVENTORY partition created by EVE. I think, if somebody want to hack something, it will do it regardless of our implementation...

Reply:
yeah, exactly. It's user input, and you never trust user :)

Reply:
Ok, after reading readme a second time, I figured that this tool is just to check what the heck we just put on the partition, and not to actually configure EVE. Is my understanding right?

However, in such case, why user can't simply open the usb stick, and check the contents in a file browser. Windows should be able to read fat32.

Reply:
Right, only for reading the partition with expected files inside.
Windows cannot recognize our partition table properly, and Inventory does not show in explorer.